reference/dic_tags.rst

```diff
@@ -470,6 +471,77 @@ To add a new rendering strategy - in addition to the core strategies like
 :class:`Symfony\\Component\\HttpKernel\\Fragment\\FragmentRendererInterface`,
 register it as a service, then tag it with ``kernel.fragment_renderer``.
 
+kernel.locale_aware
+-------------------
+
+.. versionadded:: 4.3
+
+    The ``kernel.locale_aware`` tag was introduced in Symfony 4.3.
+
+**Purpose**: To interact with the locale
+
+Setting and retrieving the locale can be done via the configuration or
+thanks to container parameters or even listeners / route parameters.
+
+Thanks to the ``Translation`` contract, the locale can be set via services.
+
+To register your own locale aware service, first create a service that implements
+the :class:`Symfony\\Contracts\\Translation\\LocaleAwareInterface` interface::
+
+    // src/Locale/MyCustomLocaleHandler.php
+    namespace App\Locale;
+
+    use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+    class MyCustomLocaleHandler implements LocaleAwareInterface
+    {
+        public function setLocale($locale)
+        {
+            // ... do some sort of operations to set the locale
```

Comment:
This setter should not be called as it used to pass the locale to the context of the class when built or when using the new switcher.
I would remove this comment and hard code a proper assignation instead:
```php
$this->locale = $locale;
```